query/src/main/java/org/infinispan/query/clustered/ClusteredQueryInvoker.java

```diff
@@ -116,26 +86,8 @@ QueryResponse unicast(Address address, SegmentsClusteredQueryCommand cmd) {
       return results;
    }
 
-   private Future<QueryResponse> localInvoke(SegmentsClusteredQueryCommand cmd) {
-      return asyncExecutor.submit(() -> {
```

Comment:
Also noticed that it seems the `asyncExecutor` can be removed from this code and subsequent callers.